src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -242,7 +256,7 @@ bool IsValidTriggerCharacterForDocument(Document document, char triggerCharacter
                     FilterText = item.FilterText,
                     Kind = GetCompletionKind(item.Tags),
                     Data = completionResolveData,
-                    Preselect = item.Rules.SelectionBehavior == CompletionItemSelectionBehavior.HardSelection,
+                    Preselect = ShouldItemBePreselected(item),
```

Comment:
so if an item should be preselected, have we considered setting things up that we ensure that the items around it are also sent down with it (so there's no weird gap between the items that are sent and the completed item?   or does this work by teh client continually pulling until it gets all the information necessary to get to that point?

Reply:
yeah that's definitely an option.  I'm not sure exactly how much to send 'around' the item though, at some point there will be a weird gap.  The client doesn't pull as you scroll unfortunately.  I could just choose a number of items to add before and after, but then the gap just moves.  I'm not sure if that would be more confusing than one preselected item at the end

Reply:
discussed a bit offline - a 1k item window approach was suggested where we take the preselected item + the 1k items around the preselected result (while keeping the preselected item as close to the middle of the 1k items as possible).  

I'll take a look and see  what that looks like in a followup